src/shared/dlt_user_shared.c

```diff
@@ -143,7 +149,30 @@ DltReturnValue dlt_user_log_out3(int handle, void *ptr1, size_t len1, void *ptr2
     iov[2].iov_base = ptr3;
     iov[2].iov_len = len3;
 
-    bytes_written = writev(handle, iov, 3);
+    /* Get Daemon FIFO size */
+    if ((daemon_fifo_size = fcntl(handle, F_GETPIPE_SZ , 0)) == -1)
+    {
+        snprintf(str,sizeof(str),"get FIFO size error, could not perform write operation: %s\n",strerror(errno));
+        dlt_log(LOG_ERR, str);
+        return DLT_RETURN_ERROR;
+    }
+    if (daemon_fifo_size > 0)
+    {
+        if (ioctl(handle, FIONREAD, &non_read_bytes) != -1)
+        {
+            // Determine the FIFO size availability prior writing data to PIPE
+            if ((daemon_fifo_size - non_read_bytes) > (len1+len2+len3))
+            {
+                bytes_written = writev(handle, iov, 3);
+            }
+            else
+            {
+                snprintf(str,sizeof(str),"Not enough space available to write the data to daemon FIFO, hence storing it in ring buffer\n");
+                dlt_log(LOG_DEBUG, str);
```

Comment:
Does not need the str here, you can log the message string directly with dlt_log(). In additional, if you need logging with format string, please use dlt_vlog().

Reply:
@votrungchi updated dlt_log with dlt_vlog as per suggestion in new commit